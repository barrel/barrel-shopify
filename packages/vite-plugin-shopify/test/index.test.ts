import { describe, expect, it, vi } from 'vitest'
import path from 'node:path'
import { build, normalizePath } from 'vite'
import shopify from '../src'
import fs from 'node:fs/promises'

describe('vite-plugin-shopify', () => {
  it('builds out .liquid files for production', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        shopify({
          themeRoot: path.join(__dirname, '__fixtures__'),
          sourceCodeDir: path.join(__dirname, '__fixtures__', 'frontend'),
          additionalEntrypoints: [
            normalizePath(path.join(__dirname, '__fixtures__', 'resources', 'js', 'foo.js'))
          ],
          snippetFile: 'vite-tag.liquid'
        })
      ],
      resolve: {
        alias: {
          '@@': normalizePath(path.resolve(path.join(__dirname, '__fixtures__', 'resources', 'js')))
        }
      }
    })

    const tagsHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-tag.liquid'), { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
  })
})

vi.mock('fast-glob', () => {
  return {
    default: {
      sync: vi.fn()
        // mock entries
        .mockReturnValueOnce([
          path.join(__dirname, '__fixtures__', 'resources', 'js', 'foo.js'),
          path.join(__dirname, '__fixtures__', 'frontend', 'entrypoints', 'theme.css'),
          path.join(__dirname, '__fixtures__', 'frontend', 'entrypoints', 'theme.ts')
        ])
    }
  }
})
