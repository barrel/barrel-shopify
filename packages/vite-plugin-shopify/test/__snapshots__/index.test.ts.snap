// Vitest Snapshot v1

exports[`vite-plugin-shopify > builds out .liquid files for production 1`] = `
"{% comment %}
  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = vite-tag | replace: '@@/', '../../resources/js/' | replace: '~/', '../' | replace: '@/', '../' %}
{% if path == \\"/test/__fixtures__/frontend/entrypoints/theme.css\\" or path == \\"theme.css\\" %}
  {{ 'theme-1a84b608.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == \\"/test/__fixtures__/frontend/entrypoints/theme.ts\\" or path == \\"theme.ts\\" %}
  <script src=\\"{{ 'theme-add44336.js' | asset_url | split: '?' | first }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
  <link rel=\\"modulepreload\\" href=\\"{{ 'modulepreload-polyfill-ec808ebb.js' | asset_url | split: '?' | first }}\\" crossorigin=\\"anonymous\\">
{% elsif path == \\"/test/__fixtures__/resources/js/foo.js\\" or path == \\"../../resources/js/foo.js\\" %}
  <script src=\\"{{ 'foo-add44336.js' | asset_url | split: '?' | first }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
  <link rel=\\"modulepreload\\" href=\\"{{ 'modulepreload-polyfill-ec808ebb.js' | asset_url | split: '?' | first }}\\" crossorigin=\\"anonymous\\">
{% endif %}
"
`;
